### ISSUE TITLE
Not getting error message when attempting to Register class with reserved name

### ISSUE THREAD

**Issue by jlk** [osohq/oso#1465]
I've got a golang type named `store.Resource`.

I've added code to register it, such as 
```
oso.RegisterClass(reflect.TypeOf(store.Resource{}), nil)
```
That line itself doesn't incur an error. But when I call `oso.LoadFiles()`, I then see the following message:
```
types.ErrorKindRuntime{RuntimeErrorVariant:types.RuntimeErrorInvalidState{Msg:"Cannot add MRO for unregistered class Resource"}}
```

At first I thought this was due to the `resource Resource` block in the polar file, but even with that removed I get the error message.

Chatting with @gj on Slack, the following workaround allows me to use the class:
```
oso.RegisterClassWithName(reflect.TypeOf(store.Resource{}), nil, "MyResource")
```

Perhaps an error message should generated in `RegisterClass()` that says that reserved keywords cannot be registered?

**Comment by gj** [comment]
I would expect us to hit this: https://github.com/osohq/oso/blob/49214bfe2cdad1e592b37b0b54a7f37874fdb30e/polar-core/src/kb.rs#L590-L596

(The `RESOURCE_UNION_NAME` constant is `"Resource"`.)

**Comment by jlk** [comment]
Perhaps because I'm passing `store.Resource{}` and not just `Resource{}`? (I realize I don't have to specify the full name in the model, but not sure if the substring happens before or after `register_constant()`...

**Comment by gj** [comment]
@jlk yeah that could definitely be it. Either way, we should shoot for parity between the class names we accept via class registration & policy loading APIs
